app/src/main/cpp/skyline/services/base_service.h

```diff
@@ -138,39 +28,48 @@ namespace skyline::service {
       protected:
         const DeviceState &state; //!< The state of the device
         ServiceManager &manager; //!< A reference to the service manager
-        std::unordered_map<u32, std::function<void(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)>> vTable; //!< This holds the mapping from a function's CmdId to the actual function
+        std::unordered_map<u32, std::function<Result(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)>> vTable; //!< This holds the mapping from a function's CmdId to the actual function
```

Comment:
We should probably replace `std::function` with a straight function pointer here, we don't use any of the extra functionality provided by it and it's actually slower to my knowledge: `Result(*)(type::KSession &, ipc::IpcRequest &, ipc::IpcResponse &)`.

Reply:
We want the name eventually

Reply:
I forgot, my bad